src/dbnode/storage/series/buffer.go

```diff
@@ -425,9 +427,33 @@ func (b *dbBuffer) Tick(blockStates map[xtime.UnixNano]BlockState, nsCtx namespa
 }
 
 func (b *dbBuffer) Bootstrap(bl block.DatabaseBlock) {
-	blockStart := bl.StartTime()
-	buckets := b.bucketVersionsAtCreate(blockStart)
-	buckets.bootstrap(bl)
+	// Only blocks that have never been flushed before are bootstrapped, so bootstrapped
+	// blocks should be treated the same as warm writes so that they get written to disk as
+	// part of the initial flush.
+	b.load(bl, WarmWrite)
+}
+
+func (b *dbBuffer) Load(bl block.DatabaseBlock) {
```

Comment:
I feel like somewhere along this call stack should be a check for whether cold writes are enabled or not, otherwise these bootstrapped blocks would not get flushed.